module/purpose/Goalie/src/Goalie.cpp

```diff
@@ -0,0 +1,102 @@
+            this->log_level    = config["log_level"].as<NUClear::LogLevel>();
+            cfg.ready_position = config["ready_position"].as<Expression>();
+        });
+
+        on<Provide<GoalieTask>, Optional<Trigger<GameState>>>().then(
+            [this](const GoalieTask& goalie_task, const std::shared_ptr<const GameState>& game_state) {
+                // Do not use GameController information if force playing or force penalty shootout
+                if (goalie_task.force_playing) {
+                    play();
+                    return;
+                }
+
+                // Check if there is GameState information, and if so act based on the current mode
+                if (game_state) {
+                    switch (game_state->data.mode.value) {
+                        case GameMode::PENALTY_SHOOTOUT: emit<Task>(std::make_unique<PenaltyShootoutGoalie>()); break;
+                        case GameMode::NORMAL:
+                        case GameMode::OVERTIME: emit<Task>(std::make_unique<NormalGoalie>()); break;
+                        default: log<NUClear::WARN>("Game mode unknown.");
+                    }
+                }
+            });
+
+        // Normal READY state
+        on<Provide<NormalGoalie>, When<Phase, std::equal_to, Phase::READY>>().then([this] {
+            emit<Task>(std::make_unique<WalkToFieldPosition>(
+                Eigen::Vector3f(cfg.ready_position.x(), cfg.ready_position.y(), 0),
+                cfg.ready_position.z()));
+        });
+
+        // Normal PLAYING state
+        on<Provide<NormalGoalie>, When<Phase, std::equal_to, Phase::PLAYING>>().then([this] { play(); });
+
+        // Normal UNKNOWN state
+        on<Provide<NormalGoalie>, When<Phase, std::equal_to, Phase::UNKNOWN_PHASE>>().then(
+            [this] { log<NUClear::WARN>("Unknown normal game phase."); });
+
+        // Default for INITIAL, SET, FINISHED, TIMEOUT
+        on<Provide<NormalGoalie>>().then([this] { emit<Task>(std::make_unique<StandStill>()); });
+
+        // Penalty shootout PLAYING state
+        on<Provide<PenaltyShootoutGoalie>, When<Phase, std::equal_to, Phase::PLAYING>>().then([this] { play(); });
+
+        // Penalty shootout UNKNOWN state
+        on<Provide<PenaltyShootoutGoalie>, When<Phase, std::equal_to, Phase::UNKNOWN_PHASE>>().then(
+            [this] { log<NUClear::WARN>("Unknown penalty shootout game phase."); });
+
+        // Default for INITIAL, READY, SET, FINISHED, TIMEOUT
+        on<Provide<PenaltyShootoutGoalie>>().then([this] { emit<Task>(std::make_unique<StandStill>()); });
+    }
+
+    void Goalie::play() {
+        // Stop the ball!
+        // Second argument is priority - higher number means higher priority
+        emit<Task>(
+            std::make_unique<WalkToFieldPosition>(Eigen::Vector3f(cfg.ready_position.x(), cfg.ready_position.y(), 0),
+                                                  cfg.ready_position.z()),
+            1);
```

Comment:
I'm surprised clang format likes this lol, surely 
```suggestion
                                                  cfg.ready_position.z()), 1);
```

Reply:
hm I see why it would do this, it's just doesn't look nice. there might be a nicer way, I'll do a new suggestion on the whole line